queries/tact/highlights.scm

```diff
@@ -94,9 +95,8 @@
 (tlb_serialization
   "as" @keyword
   type: (identifier) @type.builtin
-  (#any-of? @type.builtin
-    "int8" "int16" "int32" "int64" "int128" "int256" "int257" "uint8" "uint16" "uint32" "uint64"
-    "uint128" "uint256" "coins" "remaining" "bytes32" "bytes64"))
+  (#match? @type.builtin
```

Comment:
Sorry, but that is inacceptable. Always prefer an explicit list if possible; otherwise use `#lua-match?` and Lua patterns. If necessary, use multiple patterns instead of shoving everything into a single complicated regex.

Reply:
Sure, will fix it in a moment

Reply:
Resolved, please take a look

Reply:
Do you really need that full generality (uint69)? It's not quite clear to me what you are trying to capture here.

Reply:
In the regexes there I expressed ranges `int1` - `int256` and `uint1` - `uint256`. Could've went for `u?int[0-9][0-9]?[0-9]?`, but that would have been too general (i.e. allowing `int0` and `int999`), so here we are :)

Reply:
So are you actually expecting uint37? Such patterns are extremely performance critical, so it's important to be flexible here. Again, it's much better to have a long list of a few dozen explicit values than to have a clever but expensive pattern.

Reply:
(Obviously I know nothing about tact, so treat them as honest questions.)

Reply:
Yeah, I do. But isn't this kind of regex rather cheap? It doesn't do lookaheads (expensive) or lookbehinds (_very_ expensive), just basic `optional?` and `[single character from here]`. Not even `*` for repeats or `()` groups as before

Reply:
Regex are never cheap in this context; even Lua patterns are a different order of magnitude, and that matters a lot when you're running in a hot loop (which this very much is, being part of the main display loop).

Reply:
And if you have to, why not simply do `uint[0-9]*`? What's the harm in highlighting `uint29886`?

(Tree-sitter is used for correctly highlighting _correct_ code, not for linting incorrect code.)

Reply:
(And you need separate patterns for that; you can't chain predicates this way.)

Reply:
> (Tree-sitter is used for correctly highlighting _correct_ code, not for linting incorrect code.)

Went with that idea and just removed all the patterns there. Incorrect code would still be highlighted now, but that is fixable on the linting/diagnostics side from, say, future LSPs